src/analysis/convex/basic.lean

```diff
@@ -254,6 +255,71 @@ begin
   simp [set.mem_image, mem_add, eq_comm]
 end
 
+lemma convex.combo_to_vadd {a b : ℝ} {x y : E} (h : a + b = 1) :
+  a • x + b • y = b • (y - x) + x  :=
+  eq.symm (calc
+    b • (y - x) + x = b • (y - x) + (1 : ℝ) • x          : by rw [one_smul]
+                ... = b • (y - x) + (a + b) • x           : by rw [h]
+                ... = (b • y - b • x) + (a • x + b • x)   : by rw [add_smul, smul_sub]
+                ... = a • x + b • y                       : by abel )
+
+/--
+Applying an affine map to an affine combination of two points yields
+an affine combination of the images.
+ -/
+lemma convex.combo_affine_apply {a b : ℝ} {x y : E} {f : affine_map ℝ E E F F}
+  (h : a + b = 1) : f (a • x + b • y) = a • (f x) + b • (f y) :=
+begin
+  rw [convex.combo_to_vadd h],
+  calc
+    f (b • (y - x) + x)
+          = f.to_fun (b • ((y -ᵥ x) : E) +ᵥ x)                    : rfl
+      ... = f.linear.to_fun (b • ((y -ᵥ x) : E)) +ᵥ f.to_fun x    : by rw [f.map_vadd']; refl
+      ... = b • f.linear ((y -ᵥ x) : E) + f.to_fun x              : by rw [f.linear.map_smul']; refl
```

Comment:
We try to avoid `f.to_fun`. We have non-primed lemmas `affine_map.map_vadd`, `linear_map.map_smul`, `affine_map.linear_map_vsub` to avoid `.to_fun` in proofs.

Reply:
I see -- good to know.